lib/index.js

```diff
@@ -36,143 +64,109 @@ module.exports = {
+
+          let method = action.method.toLowerCase();
+          if (method === 'delete') { method = 'del'; }
+          if (method === 'options') { method = 'opts'; }
+
+          const fullPath = action['x-bravado-fullPath'];
+          const routePath = fullPath.replace(/\{((?:[\w.~-]|(?:%[0-9A-F]{2}))+)\}/g, ':$1');
+          if (!routes[routePath]) { routes[routePath] = []; }
+
+          routes[routePath].push({
+            name: [r, a].join('.'),
+            method: method,
             version: api.info.version,
-            handler: indexRouteHandler(api)
-          }];
-        });
-        Object.keys(routes).sort().reverse().forEach(function(routePath) {
-          routes[routePath].forEach(function(route) {
-            debug('routing %s@%s for %s', routePath, route.version, route.name);
-            if (route.method !== 'opts') {
-              server.opts(routePath,
-                cors.preflight({ extraHeaders: route.expectedHeaders }));
-            }
-            server[route.method](
-              { path: routePath, version: route.version },
-              cors.headers(),
-              route.handler
-            );
+            expectedHeaders: actionExpectedHeaders,
+            handler: routeHandler(resolved, r, a, require(controllerPath), options.context)
           });
         });
-        return q.resolve();
-      })
-      .then(function() {
-        server.on('after', function(req, res, route, err) {
-          logger(req, res, route, err);
-        });
-        server.on('NotFound', function(req, res, err, cb) {
-          res.send(404, { type: 'NotFound', message: err.message });
-          cb();
-        });
-        server.on('MethodNotAllowed', function(req, res, err, cb) {
-          res.send(405, { type: 'MethodNotAllowed', message: err.message });
-          cb();
-        });
-        server.on('VersionNotAllowed', function(req, res, err, cb) {
-          res.send(400, { type: 'VersionNotAllowed', message: err.message });
-          cb();
-        });
-        server.on('InternalServer', function(req, res, err, cb) {
-          res.send(500, { type: 'ServerError', message: err.message });
-          cb();
-        });
-        server.on('uncaughtException', function(req, res, route, err) {
-          logger(req, res, route, err);
-          if (!res.headersSent) {
-            res.send(err.statusCode || 500, { type: err.type || 'Error', message: err.message });
-          }
-        });
+      });
+      routes[api.basePath] = [{
+        name: 'index',
+        method: 'get',
+        version: api.info.version,
+        handler: indexRouteHandler(api)
+      }];
+    });
 
-        server.listen(
-          config.get('port'),
-          config.get('host'),
-          function() {
-            debug('%s listening at %s', server.name, server.url);
-          }
+    keys(routes).forEach((routePath) => {
```

Comment:
why did you drop the `sort().reverse()`?

Reply:
took it off while testing and forgot about him - my bad.